app/src/main/java/pm/gnosis/heimdall/ui/qrscan/Videographer.kt

```diff
@@ -0,0 +1,182 @@
+package pm.gnosis.heimdall.ui.qrscan
+
+import android.app.Activity
+import android.graphics.ImageFormat
+import android.graphics.SurfaceTexture
+import android.hardware.Camera
+import android.os.Build
+import android.os.Handler
+import android.os.HandlerThread
+import android.view.Surface
+import android.view.TextureView
+import timber.log.Timber
+
+/*
+ * Check https://github.com/walleth/walleth/tree/master/app/src/main/java/org/walleth/activities/qrscan
+ */
+@Suppress("DEPRECATION")
+class Videographer(val activity: Activity) {
+    private val scanner = Scanner(this)
```

Comment:
What about injecting this (and maybe even the camera) that would allow us to do some easier testing

Reply:
Also using our own interface for the camera would allow us to add support for Camera2 api later on. For this the necessary methods should expose observables (not sure if you want to do this in this PR)

Reply:
The problem with this injection is that there's a circular dependency also on the scanner side. `Videographer` depends on the `Scanner` and the `Scanner` depends on the `Videographer`. I'll look into it in more detail.